WordPress/gutenberg
# Fix toolbar alignment in widget block editor

## Description
Fixes https://github.com/WordPress/gutenberg/issues/31021.

Don't apply the `has-parent` CSS class to the toolbar if the `__experimentalParentSelector` flag is set. This class was applying the left margin to the toolbar when a block was selected in the widget area.

## How has this been tested?

1. In _Appearance_ > _Widgets_, select a top level block in a widget area and ensure toolbar alignment is correct.
2. Add the Navigation block to a page.
3. Click _Add all pages_.
4. Select the inner Page List block and ensure the `block-editor-block-contextual-toolbar` element has the `has-parent` class, and that the toolbar alignment is correct.

## Screenshots 
![Screen Shot 2021-05-11 at 7 33 52 AM](https://user-images.githubusercontent.com/1190420/117808755-453e9800-b22b-11eb-9497-92061f05b5ee.jpg)

## Types of changes
Bug fix

## Checklist:
